cpp/ps0004/p4Solution.cpp

```diff
@@ -1,22 +1,25 @@
 #include<iostream>
-using namespace std;
 
+using namespace std;
 
 int main() {
+	// prompting user for number 1.5/1.5
 	float num;
 	char character;
 	cout << "Enter the ascii code[32 : 126] to get its corresponding character: ";
 	cin >> num;
+
+	// checking invalidaty of user's input 4/4
 	if (num == 0)
 		cout << "Invalid character/s or Zero value";
 	else if (num != int(num))
 		cout << "In valid: a floating-point number\n";
-	else if (num < 0 or num >126 or !(num >= 32 and num <= 126))
+	else if (!(num >= 32 && num <= 126))
 		cout << "Invalid: Out of range\n";
 	else {
 		character = static_cast <char>(num);
 		cout << "The Character of Ascii code: " << num << " is " << character << endl;
 	}

```

Comment:
- [x] used `if` structure in a right way
- [x] `if` structure take in consideration all invalid cases
- [x] all expressions to check invalidity are correct
- [x] used `std::cout` and `std::endl` or `\n` to print a different message for the 3 cases